libs/sdk-common/src/input_parser.rs

```diff
@@ -183,8 +191,21 @@ use crate::prelude::*;
 pub async fn parse(
     input: &str,
     external_input_parsers: Option<&[ExternalInputParser]>,
+    dns_resolver: Option<&AsyncResolver<GenericConnector<TokioRuntimeProvider>>>,
```

Comment:
Do you see a need to provide a "custom" dns resolver to `parse`? 

The interface would be leaner if a `parse` caller didn't need to provide one. We can create a new one on every call.

Reply:
I agree with you. I think that I misunderstood this comment to the liquid PR (https://github.com/breez/breez-sdk-liquid/pull/619/files/51cf0e48605d5df88f26a3b12a21e2332c89e607)

Reply:
Not sure about re-creating it on every call, since it's something that we expect to be frequently evoked from the user's side. 
I'm not against instantiating the resolver internally altogether, you could try taking a look at the `lazy_static` crate and starting it there.